.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateParser.cs

```diff
@@ -592,6 +592,45 @@ private DateTimeResolutionResult ParseImplicitDate(string text, DateObject refer
                 return ret;
             }
 
+            /*
+             Adding support to parse addtitonal Implicit date references under tasksmode.
+            eg next week will get mapped to same day of next week,
+            next month will get mapped to starting day of comming month,
+            next year will get mapped to starting date of coming year.
+
+            Input text : meet me next week (refrence time 01-08-2022)
+            Tasksmode: next week --> 08-08-2022 datetime type: date
+            Default mode: next week --> 08-08-2022 - 15-08-2022 datetime type: daterange
+             */
+            if ((config.Options & DateTimeOptions.TasksMode) != 0)
+            {
+                trimmedText = text.Trim();
```

Comment:
Could this be a method in the mode-specific code file?

Reply:
I have defined a different method for this mode specific date parsing, But I don't think it's possible to have this method in mode specific file.